packages/assets-controllers/src/TokenBalancesController.ts

```diff
@@ -442,12 +443,28 @@ export class TokenBalancesController extends StaticIntervalPollingController<{
 
   /**
    * Override to handle our custom polling approach
+   *
+   * @param tokenSetId - The token set ID to stop polling for
    */
-  override _stopPollingByPollingTokenSetId() {
-    this.#isControllerPollingActive = false;
-    this.#requestedChainIds = []; // Clear original intent when stopping
-    this.#intervalPollingTimers.forEach((timer) => clearInterval(timer));
-    this.#intervalPollingTimers.clear();
+  override _stopPollingByPollingTokenSetId(tokenSetId: PollingTokenSetId) {
+    const parsedTokenSetId = JSON.parse(tokenSetId);
+    const chainsToStop = parsedTokenSetId.chainIds || [];
+
+    // Compare with current chains - only stop if it matches our current session
+    const currentChainsSet = new Set(this.#requestedChainIds);
+    const stopChainsSet = new Set(chainsToStop);
+
+    // Check if this stop request is for our current session
+    const isCurrentSession =
+      currentChainsSet.size === stopChainsSet.size &&
+      [...currentChainsSet].every((chain) => stopChainsSet.has(chain));
```

Comment:
Okay I think I get it.

However what is a user want to only turn off some chains but not all?

Reply:
This was confusion on my side, resolving

Reply:
Dev notes:

This _stopPollingByPollingTokenSetId was stopping polling for all chains, now we only stop when the chains to stop are identical.

When can this happen? Pretty much _never_, any time you change chains, this is always different. We are only keeping this logic as a fallback for the current websocket work.

Code flow:
1. We start polling (add to our polling list)
    - Internally this handles stop polling for chains we're are not listening to anymore.
2. This override is called.